Lib/idlelib/debugger_r.py

```diff
@@ -299,7 +299,22 @@ def __init__(self, conn, shell, oid):
         self.conn = conn
         self.shell = shell
 
-    def call(self, methodname, *args, **kwargs):
+    def call(*args, **kwargs):
+        if len(args) >= 2:
+            self, methodname, *args = args
+        elif not args:
+            raise TypeError("descriptor 'call' of 'IdbProxy' object "
+                            "needs an argument")
+        elif 'methodname' in kwargs:
+            methodname = kwargs.pop('methodname')
+            self, *args = args
+            import warnings
+            warnings.warn("Passing 'methodname' as keyword argument is deprecated",
+                          DeprecationWarning, stacklevel=2)
+        else:
+            raise TypeError('call expected at least 1 positional argument, '
+                            'got %d' % (len(args)-1))
+
```

Comment:
I consider the removal of methodname a regression.  Please don't.  This, like most of idlelib, is private implementation.   As near as I can tell only one instance, idb_proxy, is created (at a time) and passed to Debugger as idb.  I cannot find anywhere that idb or idb.call is called within Debugger.

Reply:
There is no a regression (`IdbProxy.call()` is never called with *methodname* passed as a keyword argument). But I agree that this deprecation is not necessary because `IdbProxy.call()` is called only in few places in this file, and only with positional arguments. And it is not public.

We could just add `/` after `methodname` in future for the case if this method will be called with arbitrary keyword arguments.

Reply:
'regression' refers to the signature itself. As I said in my comment on the  WIP, signatures should communicate requirements and '*args'  fails in this regard.  (Which is why I think we should adding / would be good.)